### PR TITLE
feat: created followers instance

### PR DESCRIPTION
**Pull request by fominmaksim**
- created follower instance

### CODE REVIEW

File: src/repositories/followers.repository.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { FollowerModel } from "../models/Follower.model";
+
+export const listFollowers = (queryParams: object) => {
```
Comment (AndrewKaenko) [inline src/repositories/followers.repository.ts]: the followers list returns a Promise. These functions should be async

File: src/models/Follower.model.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { ObjectId } from "mongodb";
+import mongoose, { Schema, Document } from "mongoose";
+import paginate from "mongoose-paginate-v2";
+
+export interface Follower extends Document {
+  followerId: ObjectId;
+  ownerId: ObjectId;
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
  followerId: Types.ObjectId;
  ownerId: Types.ObjectId;
```
https://mongoosejs.com/docs/typescript.html

File: src/models/Follower.model.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { ObjectId } from "mongodb";
+import mongoose, { Schema, Document } from "mongoose";
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
import mongoose, { Schema, Document, Types } from "mongoose";
```

File: src/models/Follower.model.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+export interface Follower extends Document {
+  followerId: ObjectId;
+  ownerId: ObjectId;
+  createdAt: string;
+  updatedAt?: string;
+}
+
+const FollowerSchema: Schema = new Schema<Follower>({
+  followerId: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
    type: Schema.Types.ObjectId,
```
https://mongoosejs.com/docs/typescript.html

File: src/models/Follower.model.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    required: true,
+    ref: "User",
+  },
+
+  ownerId: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
    type: Schema.Types.ObjectId,
```

File: src/models/Follower.model.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    required: true,
+    ref: "User",
+  },
+
+  createdAt: {
+    type: String,
+    required: true,
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
default: new Date().toISOString(),
```

File: src/models/Follower.model.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { ObjectId } from "mongodb";
+import mongoose, { Schema, Document } from "mongoose";
+import paginate from "mongoose-paginate-v2";
+
+export interface Follower extends Document {
+  followerId: ObjectId;
+  ownerId: ObjectId;
+  createdAt: string;
```
Comment (VitaliiDenysiuk) [inline src/models/Follower.model.ts]:
```suggestion
  createdA?t: string;
```
We did not ask that, we filled it by default

File: src/repositories/followers.repository.ts (hunk 1 above)
Comment (VitaliiDenysiuk) [reply src/repositories/followers.repository.ts]: @AndrewKaenko No. We set async at setvice

File: src/repositories/followers.repository.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  const query = {};
+  const options = {
+    ...queryParams,
+  };
+  return FollowerModel.paginate(query, options);
+};
```
Comment (VitaliiDenysiuk) [inline src/repositories/followers.repository.ts]:
```suggestion
export const listFollowers = (query: object, options: object) => {
   return FollowerModel.paginate(query, options);
};
```
check tweet repository

File: src/routes/followers.route.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  deleteFollower,
+  getFollowers,
+} from "../services/followers.service";
+
+const followersRouter = express.Router();
+
+followersRouter
+  .route("/")
+  .get(isAuthorized, async (req: Request, res: Response) => {
+    const result = await getFollowers(req.query);
```
Comment (VitaliiDenysiuk) [inline src/routes/followers.route.ts]:
```suggestion
    const result = await getFollowers(new PaginationParameters(req));
```
 tweet route for example

File: src/routes/followers.route.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+followersRouter
+  .route("/")
+  .post(
+    isAuthorized,
+    validation(followersSchema),
+    async (req: Request, res: Response) => {
+      const { followerId, ownerId } = req.body;
+      await addFollower(followerId, ownerId);
+      res.sendStatus(200);
```
Comment (VitaliiDenysiuk) [inline src/routes/followers.route.ts]:
```suggestion
      const newFollower = await addFollower(followerId, ownerId);
      res.status(201).send(newFollower);
```
tweet example

File: src/services/followers.service.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import {
+  addOneFollower,
+  deleteFollowerById,
+  listFollowers,
+} from "../repositories/followers.repository";
+
+export const getFollowers = async (query: object) => {
+  const followersList = await listFollowers(query);
+  return followersList;
+};
```
Comment (VitaliiDenysiuk) [inline src/services/followers.service.ts]:
```suggestion
export const getFollowers = async (query: PaginationParameters<never, never>) => {
  const followersList = await listFollowers(...query.get());
  return followersList;
};
```
tweet exapmle

File: src/services/followers.service.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+export const addFollower = async (ownerId: string, followerId: string) => {
```
Comment (VitaliiDenysiuk) [inline src/services/followers.service.ts]:
```suggestion
export const addFollower =  (ownerId: string, followerId: string) => {
```
not use async without await

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by VitaliiDenysiuk — APPROVED** [review]
looks good to me

**Review by AndrewKaenko — APPROVED** [review]
Looks perfect to me!

**Pull request merged**